pkg/strategy/elliottwave/strategy.go

```diff
@@ -0,0 +1,495 @@
+package elliottwave
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/c9s/bbgo/pkg/bbgo"
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/indicator"
+	"github.com/c9s/bbgo/pkg/strategy"
+	"github.com/c9s/bbgo/pkg/types"
+	"github.com/c9s/bbgo/pkg/util"
+	"github.com/sirupsen/logrus"
+)
+
+const ID = "elliottwave"
+
+var log = logrus.WithField("strategy", ID)
+var Two fixedpoint.Value = fixedpoint.NewFromInt(2)
+var Three fixedpoint.Value = fixedpoint.NewFromInt(3)
+var Four fixedpoint.Value = fixedpoint.NewFromInt(4)
+var Delta fixedpoint.Value = fixedpoint.NewFromFloat(0.00001)
+
+func init() {
+	bbgo.RegisterStrategy(ID, &Strategy{})
+}
+
+type SourceFunc func(*types.KLine) fixedpoint.Value
+
+type Strategy struct {
+	Symbol string `json:"symbol"`
+
+	bbgo.StrategyController
+	types.Market
+	strategy.SourceSelector
```

Comment:
this does not need to be embedded in the struct? we can use a field for this?

Reply:
it could be, just that when we call `getSource`
there's no syntax sugar like `s.getSource`. Need to write `s.Selector.getSource`.